contracts/integrations/contracts/dydx/ZeroExV3MultiOrderExchangeWrapper.sol

```diff
@@ -112,11 +128,21 @@ contract ZeroExV2MultiOrderExchangeWrapper is
         LibOrder.Order[] memory orders = parseOrders(orderData, makerToken, takerToken);
         bytes[] memory signatures = parseSignatures(orderData);
 
-        // ensure that the exchange can take the takerTokens from this contract
+        // ensure that the 0x exchange can take the takerTokens from this contract
         takerToken.ensureAllowance(ZERO_EX_TOKEN_PROXY, requestedFillAmount);
 
+        // make sure that the protocol fee collector can take WETH from this contract
+        IExchange v3Exchange = IExchange(ZERO_EX_EXCHANGE);
+        address protocolFeeCollector = v3Exchange.protocolFeeCollector();
```

Comment:
I'm not a fan of the fact that the `protocolFeeCollector` will be approved redundantly here. I think that a better way to handle this approval would be to expose a public endpoint that approves the `protocolFeeCollector` endpoint from the registered exchange, so that the staking contract only needs to be approved once every upgrade.